### PR TITLE
updated AMP Cache URL with new algorithm 

### PR DESCRIPTION
**Pull request by CrystalOnScript**
Resolves issue [#3851](https://github.com/ampproject/amp.dev/issues/3851)

### CODE REVIEW

File: pages/content/amp-dev/documentation/guides-and-tutorials/learn/amp-caches-and-cors/amp-cache-urls.md [hunk 1]
```diff
@@ -207,8 +204,9 @@ The rest of the algorithm assumes that the “domain prefix” contains at least
...
-1.  Punycode encode the result of Step 2. See [RFC 3492](https://tools.ietf.org/html/rfc3492) for punycode.
+1.  If the domain prefix starts with "`0-`" and ends with "`-0`", strip both the "`0-`" prefix and the "-0" suffix.
+1.  Iterate through the characters output by Step 2 in order, emitting them as encountered. When you encounter a "`-`" (hyphen), peek at the following character. If the following character is also a "`-`" (hyphen), skip both characters from the input and emit a single "`-`" (hyphen). If the following character is any other character, skip only the current single "`-`" (hyphen) and emit a "`.`" (dot).  For example, `a--b-example-com` becomes `a-b.example.com`.
+1.  Punycode encode the result of Step 3. See [RFC 3492](https://tools.ietf.org/html/rfc3492) for punycode.
 
 The result of Step 3 will be the Publisher Domain. The protocol is unavailable from the domain itself, but is either `http` or `https`. The port is always the default for the protocol.
```
Comment (Gregable) [inline pages/content/amp-dev/documentation/guides-and-tutorials/learn/amp-caches-and-cors/amp-cache-urls.md:211]: Should now be Step 4.

**Comment by Gregable** [comment]
Thanks!

**Comment by CrystalOnScript** [comment]
Sorry, merged too quickly - will fix that pesky step 3.

**Comment by CrystalOnScript** [comment]
got it
https://github.com/ampproject/amp.dev/commit/23eb10e5e7343237a2df68114e90b220df3e34df

**Pull request merged**